packages/hardhat-core/test/internal/core/config/config-resolution.ts

```diff
@@ -37,8 +38,8 @@ describe("Config resolution", () => {
         assert.isUndefined(config.solidity.compilers[0]?.settings?.evmVersion);
         assert.equal(config.defaultNetwork, "hardhat");
 
-        const hardhatNetworkConfig: HardhatNetworkUserConfig = config.networks
-          .hardhat as HardhatNetworkUserConfig;
+        const hardhatNetworkConfig: HardhatNetworkConfig = config.networks
+          .hardhat as HardhatNetworkConfig;
```

Comment:
Another change related to the difference between the type of `minGasPrice` in the user type and the resolved type. This is the correct type anyway, it's just that it wasn't a problem before.